### DISCUSSION TITLE
Service requiring authentification

### DISCUSSION THREAD

**Discussion by mveea** [Kong/kong#9912] [Help]
Hello,
I'm posting here a question I couldn't find an answer to in the official documentation.
I would like to use Kong in top of a service that requires authentification.
Is there a way, when we add a new service, to provide authentification information that needs to be used for this particular service.
From a higher point of view, the final client would have an account attached to kong. When he send a request to the final service through kong, his account is been authenticate against kong, then kong use the pre-defined service account to send the request to the service.

**Comment by StarlightIbuki** [comment]
I'm not sure what kind of authentification you want to use.
Authentication can be done with a custom plugin. You can modify the request per the requirement of authentication, like adding headers and parameters, signing the request, etc.
Usually, Kong as a gateway acts as the interface of trusted and untrusted, and it should be Kong to authenticate coming requests.
